### PR TITLE
Support for Custom GCLB SSL Policy

### PR DESCRIPTION
**Pull request by reisbel**
Added support for custom Google Cloud Load Balance SSL Policies.

Usage:

Example - Disable TLS 1.0 and TLS 1.1. Also disable the following ciphers:
```
TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA
TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA
TLS_RSA_WITH_3DES_EDE_CBC_SHA
TLS_RSA_WITH_AES_128_CBC_SHA
TLS_RSA_WITH_AES_128_GCM_SHA256
TLS_RSA_WITH_AES_256_CBC_SHA
TLS_RSA_WITH_AES_256_GCM_SHA384
```

Create a file with the custom SSL Policy CIDR ranges, update it with your custom values.

```bash
cat > selkies-cluster-min.auto.tfvars <<EOF
custom_ssl_policy_enabled           = true
custom_ssl_policy_delete            = false
ssl_policy_min_tls_version          = "TLS_1_2"
ssl_policy_profile                  = "CUSTOM"
ssl_policy_custom_features          = ["TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA","TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA","TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256","TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA","TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384","TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256","TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256","TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384","TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256"]
EOF
```

Execute `terraform plan` from the `setup/infra` folder to inspect the proposed changes

```bash
(cd setup/infra && gcloud builds submit --project ${PROJECT_ID?} --substitutions=_ACTION=plan )
```

Execute `terraform apply` from the `setup/infra` folder to apply the changes

```
(cd setup/infra && gcloud builds submit --project ${PROJECT_ID?} --substitutions=_ACTION=apply )
```

See below [SSL Server Test](https://www.ssllabs.com/ssltest/analyze.html) output after applying the above configuration

image

### CODE REVIEW

**Comment by danisla** [comment]
I wasn't able to find a way to delete the policy and remove it from the target proxy by setting `custom_ssl_policy_enabled = false`

My suggestion it to make removable a 2 pass process like this:

```hcl
resource "google_compute_ssl_policy" "ssl_policy" {
  name            = "${var.name}-ssl-policy"
  min_tls_version = var.ssl_policy_min_tls_version
  profile         = var.ssl_policy_profile
  custom_features = var.ssl_policy_profile == "CUSTOM" ? var.ssl_policy_custom_features : null
  count           = var.custom_ssl_policy_delete ? 0 : 1
}
```

```hcl
variable "custom_ssl_policy_delete" {
  description = "set this in 2-pass ssl_policy removal after running with custom_ssl_policy_enabled = false to remove the ssl_policy resource without dependency issues with the target proxy"
  default     = true
}
```

File: setup/infra/lb.tf [hunk 1]
```diff
@@ -129,6 +129,15 @@ resource "google_compute_target_https_proxy" "ingress" {
   name             = "istio-ingressgateway"
   url_map          = google_compute_url_map.ingress.self_link
   ssl_certificates = concat(list(google_compute_managed_ssl_certificate.ingress.self_link), values(google_compute_managed_ssl_certificate.extras).*.self_link)
+  ssl_policy = var.custom_ssl_policy_enabled ? concat(google_compute_ssl_policy.ssl_policy.*.id, [""])[0] : null
```
Comment (danisla) [inline setup/infra/lb.tf:132]: formatting. run `terraform fmt lb.tf`

Comment (reisbel) [reply setup/infra/lb.tf]: Done [here](https://github.com/selkies-project/selkies/pull/12/commits/47d9f85b641b337d3ec3b0740af57e56168a0a28)

**Comment by reisbel** [comment]
Thanks, @danisla I updated the implementation with your recommendation.

**Pull request merged**